examples/graphbolt/disk_based_feature/node_classification_offline.py

```diff
@@ -0,0 +1,644 @@
+    parser.add_argument("--enable-inference", action="store_true")
+    return parser.parse_args()
+
+
+def main():
+    start = time.time()
+    torch.set_float32_matmul_precision(args.precision)
+    if not torch.cuda.is_available():
+        args.mode = "cpu-cpu-cpu"
+    print(f"Training in {args.mode} mode.")
+    args.graph_device, args.feature_device, args.device = args.mode.split("-")
+    args.overlap_feature_fetch = args.feature_device == "pinned"
+    # For now, only sample_layer_neighbor is faster with this option
+    args.overlap_graph_fetch = (
+        args.sample_mode == "sample_layer_neighbor"
+        and args.graph_device == "pinned"
+    )
+
+    """
+    Load and preprocess on-disk dataset.
+    We inspect the in_memory field of the feature_data in the YAML file and modify
+    it to False. This will make sure the feature_data is loaded as DiskBasedFeature.
+    """
+    print("Loading data...")
+    disk_based_feature_keys = None
+    if args.cpu_cache_size_in_gigabytes > 0:
+        disk_based_feature_keys = [("node", None, "feat")]
+
+    dataset = gb.BuiltinDataset(args.dataset, root=args.root)
+    if disk_based_feature_keys is None:
+        disk_based_feature_keys = set()
+    for feature in dataset.yaml_data["feature_data"]:
+        feature_key = (feature["domain"], feature["type"], feature["name"])
+        # Set the in_memory setting to False without modifying YAML file.
+        if feature_key in disk_based_feature_keys:
+            feature["in_memory"] = False
+    dataset = dataset.load()
+
+    # Move the dataset to the selected storage.
+    graph = (
+        dataset.graph.pin_memory_()
+        if args.graph_device == "pinned"
+        else dataset.graph.to(args.graph_device)
+    )
+    features = (
+        dataset.feature.pin_memory_()
+        if args.feature_device == "pinned"
+        else dataset.feature.to(args.feature_device)
+    )
+
+    train_set = dataset.tasks[0].train_set
+    valid_set = dataset.tasks[0].validation_set
+    test_set = dataset.tasks[0].test_set
+    all_nodes_set = dataset.all_nodes_set
+    args.fanout = list(map(int, args.fanout.split(",")))
+    num_classes = dataset.tasks[0].metadata["num_classes"]
+
+    """
+    If the CPU cache size is greater than 0, we wrap the DiskBasedFeature to be
+    a CPUCachedFeature. This internally manages the CPU feature cache by the
+    specified cache replacement policy. This will reduce the amount of data
+    transferred during disk read operations for this feature.
+    
+    Note: It is advised to set the CPU cache size to be at least 4 times the number
+    of sampled nodes in a mini-batch, otherwise the feature fetcher might get into
+    a deadlock, causing a hang.
+    """
+    if args.cpu_cache_size_in_gigabytes > 0 and isinstance(
+        features[("node", None, "feat")], gb.DiskBasedFeature
+    ):
+        features[("node", None, "feat")] = gb.CPUCachedFeature(
```

Comment:
```suggestion
        features[("node", None, "feat")] = features[("node", None, "feat")].read_into_memory()
        features[("node", None, "feat")] = gb.CPUCachedFeature(
```

Reply:
You can have this line to debug, that way you won't be affected by any potential bugs inside DiskBasedFeature, it might not be handling exceptions correctly.

Reply:
That way, the error will be inside TorchBasedFeature if the rest of your code has a bug.